File: app/src/main/java/com/supercilex/robotscouter/ui/DeleteTeamDialog.java
package com.supercilex.robotscouter.ui;

import android.app.AlertDialog;
import android.app.Dialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.annotation.NonNull;
import android.support.v4.app.FragmentManager;

import com.supercilex.robotscouter.R;
import com.supercilex.robotscouter.data.model.Team;

public class DeleteTeamDialog extends DialogBase implements AlertDialog.OnClickListener {
    private static final String TAG = "DeleteTeamDialog";

    public static void show(FragmentManager manager, Team team) {
        DeleteTeamDialog dialog = new DeleteTeamDialog();
        dialog.setArguments(team.getBundle());
        dialog.show(manager, TAG);
    }

    @NonNull
    @Override
    public Dialog onCreateDialog(Bundle savedInstanceState) {
        return new AlertDialog.Builder(getContext())
                .setTitle(R.string.confirm_deletion)
                .setPositiveButton(R.string.delete, this)
                .setNegativeButton(android.R.string.no, null)
                .create();
    }

    @Override
    public void onClick(DialogInterface dialog, int which) {
        Team.getTeam(getArguments()).delete();
    }
}

File: feature/teams/src/main/java/com/supercilex/robotscouter/feature/teams/DeleteTeamDialog.kt
package com.supercilex.robotscouter.feature.teams

import android.app.Dialog
import android.os.Bundle
import android.support.v4.app.FragmentManager
import android.text.SpannableStringBuilder
import android.view.View
import androidx.core.text.bold
import com.supercilex.robotscouter.core.data.getTeamList
import com.supercilex.robotscouter.core.data.isSingleton
import com.supercilex.robotscouter.core.data.model.trash
import com.supercilex.robotscouter.core.data.toBundle
import com.supercilex.robotscouter.core.model.Team
import com.supercilex.robotscouter.core.ui.BottomSheetDialogFragmentBase
import com.supercilex.robotscouter.core.ui.show
import com.supercilex.robotscouter.core.unsafeLazy
import kotlinx.android.synthetic.main.dialog_delete_team.*

internal class DeleteTeamDialog : BottomSheetDialogFragmentBase(), View.OnClickListener {
    override val containerView: View by unsafeLazy {
        View.inflate(context, R.layout.dialog_delete_team, null)
    }
    private val teams: List<Team> by unsafeLazy { arguments!!.getTeamList().sorted() }

    override fun onDialogCreated(dialog: Dialog, savedInstanceState: Bundle?) {
        message.text = run {
            val message = resources.getQuantityString(
                    R.plurals.team_delete_message,
                    teams.size,
                    teams.first()
            )

            if (teams.isSingleton) return@run message

            val deletedTeams = SpannableStringBuilder(message).append('\n')
            for ((i, team) in teams.withIndex()) {
                deletedTeams.bold {
                    append('\n').append("${i + 1}. ")
                }.append(team.toString())
            }
            deletedTeams
        }
        delete.setOnClickListener(this@DeleteTeamDialog)
    }

    override fun onClick(v: View) {
        for (team in teams) team.trash()
        dismiss()
    }

    companion object {
        private const val TAG = "DeleteTeamDialog"

        fun show(manager: FragmentManager, teams: List<Team>) =
                DeleteTeamDialog().show(manager, TAG, teams.toBundle())
    }
}
